src/mscorlib/src/System/Diagnostics/Eventing/EventSource_CoreCLR.cs

```diff
@@ -48,8 +41,12 @@ public static void SetCurrentThreadActivityId(Guid activityId)
             // Note we can't access m_throwOnWrites because this is a static method.  
 
 #if FEATURE_PERFTRACING
-            s_currentThreadActivityId = activityId;
-#elif PLATFORM_WINDOWS
+            EventPipeInternal.EventActivityIdControl(
```

Comment:
On Windows with FEATURE_PERFTRACING on, could this, with the subsequent call to ETW result in a state where EventPipe and ETW report different activity IDs?

Reply:
It is possible that these can get out of sync if code in the process changes ETW directly.  That being said, you'll only notice this if you're using both ETW and EventPipe and comparing the two.  One thing that we could do is on Windows, always use the Windows API to get and set the activity ID.  The unfortunate part of this is that this results in a syscall into the kernel on every event write, which will be quite painful.

I also believe that managed Activities don't query the OS - they hold activity IDs in managed data structures, so I believe they are susceptible to the same problem.

Reply:
That makes sense, but what prompted me to ask that question was the use of the same variable to pass in to both functions. As I understand it using EVENT_ACTIVITY_CTRL_GET_SET_ID will change the ref passed variable to the old value of the activity ID from EventPipe. That variable would then be passed to the ETW call, meaning that the ETW has the ID that was previous set in EventPipe, not what it was intended to be set to. This is likely more relevant below than here.

Reply:
Ah... I get it.  Good catch!  I have fixed this and will push the change once the CI runs complete so that I don't throw away the current runs/results.

Reply:
One cool feature I realised recently is that you can click the ❌ or ✔next to a past commit to see it's CI results. (I also just found out that you can open an emoji keyboard on Windows 10 with Win+. but that it much less useful)

Reply:
Ah, thanks.  You learn something new every day.